lib/slack/web/faraday/response/raise_error.rb

```diff
@@ -15,6 +15,16 @@ def on_complete(env)
             error_class ||= Slack::Web::Api::Errors::SlackError
             raise error_class.new(error_message, env.response)
           end
+
+          def call(env)
+            super
+          rescue Slack::Web::Api::Errors::SlackError, Slack::Web::Api::Errors::TooManyRequestsError
+            raise
+          rescue ::Faraday::ParsingError
+            raise Slack::Web::Api::Errors::ParsingError.new('parsing_error', env.response)
```

Comment:
I would sink `parsing_error` message into the error definition itself when calling `super`

Reply:
Not sure what do you mean here.

Reply:
Like this. I don't think there's some kind of convention that the first argument must be a string, but I could be wrong? This is not a must have change for me here.

```ruby
raise Slack::Web::Api::Errors::ParsingError.new(env.response)
```

Reply:
As a user of this gem I'm expecting first argument (i. e. message) of `Slack::Web::Api::Errors::Error` to be a kinda meaningful `String`, while the second is response, that's how it works everywhere currently. So I assume that there actually is an implicit convention.
And while it will be more appropriate to catch & process `Slack::Web::Api::Errors::ParsingError` separately — I assume something like
```ruby
rescue Slack::Web::Api::Errors::Error => e
  raise unless e.message == 'parsing_error'
```
also is expected to work.

But I also don't have strong feelings about this.

Reply:
My issue is that it's an argument to the constructor that hides the message argument. I was thinking like so:

```ruby
class TimeoutError < HttpRequestError
  def initialize(response)
     super 'timeout_error'
  end
end
```

```ruby
raise Slack::Web::Api::Errors::TimeoutError
```